src/MediatR/Pipeline/PipelinePriorityAttribute.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+namespace MediatR.Pipeline
+{
+    /// <summary>
+    /// Controls the order of execution for <see cref="IPipelineBehavior{TRequest,TResponse}"/>
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
+    public class PipelinePriorityAttribute : Attribute
+    {
+        public PipelinePriorityAttribute(int priority)
+        {
+            Priority = priority;
+        }
+
+        /// <summary>
+        /// Priority of execution. Higher the the number the higher the priority. See <see cref="PipelinePriorityOrder"/> for predefined priorities.
+        /// </summary>
+        public int Priority { get; set; }
+    }
+
+    public static class PipelinePriorityOrder
```

Comment:
Should we move this out into its own file?

Reply:
We certainly can. Do you want to keep it in the same namespace?

Reply:
I think it is sensible to keep it in the same namespace, otherwise I would move all priority stuff into a `MediatR.Priority` namespace for neatness... but I think that is a community call